modules/pam_timestamp/pam_timestamp.c

```diff
@@ -816,6 +816,7 @@ main(int argc, char **argv)
 	pwd = getpwuid(getuid());
 	if (pwd == NULL) {
 		retval = 4;
+		return retval;
```

Comment:
If the user contains invalid characters, line 833 would print an error message.

Looking at the whole function, my guess (since I am not a maintainer) is, that it would make sense to move the whole pwd-processing into its own else-block right after line 819.

After the else, retval should be checked for value 4 and if it's true, then print the error and call return as suggested. This would lead to a central code point which covers the other `retval = 4` case in 827.